src/engine/Marabou.cpp

```diff
@@ -30,33 +30,23 @@
 #undef ERROR
 #endif
 
-Marabou::Marabou()
-    : _acasParser( NULL )
-    , _onnxParser( NULL )
+Marabou::Marabou( InputQuery& inputQuery )
+    : _inputQuery( inputQuery )
     , _engine()
 {
 }
 
 Marabou::~Marabou()
 {
-    if ( _acasParser )
-    {
-        delete _acasParser;
-        _acasParser = NULL;
-    }
-
-    if ( _onnxParser )
-    {
-        delete _onnxParser;
-        _onnxParser = NULL;
-    }
```

Comment:
I'm not quite sure if I need to free the `_inputQuery` object here? Would appreciate the input of a slightly better C++ programmer!  (and if so what is the best syntax to use?)